worker/provisioner/provisioner_task.go

```diff
@@ -1275,8 +1274,7 @@ func (task *provisionerTask) markMachineFailedInAZ(machine apiprovisioner.Machin
 	}
 
 	// Check if there are any zones left to try (that also match constraints).
-	zoneMap := azMachineFilterSort(task.availabilityZoneMachines)
-	for _, zoneMachines := range zoneMap {
+	for _, zoneMachines := range task.availabilityZoneMachines {
```

Comment:
we don't need to sort anymore?

Reply:
We were actually never sorting (we'd need to call `sort.Sort(zoneMap)`). The type conversion was leftover from a previous refactor, when we were calling a `.Filter()` method on the type, so it had to be the `azMachineFilterSort` type.